### PR TITLE
switch to manylinux_2_28

### LINKED ISSUE #93: C++20 might break manylinux2014
If not with the next vroom version, it'll only be a matter of time until gcc10 will run out of supported C++20 features vroom is using. Currently its running CI with gcc11. I'd suggest to move to [`manylinux_2_28`](https://github.com/pypa/manylinux#manylinux_2_28-almalinux-8-based) eventually, which has gcc12 and is basically CentOS8. I'm working quite a bit on bindings stuff these days and we're using that already, so I could this here too. WDYT @jonathf ?

### PR DESCRIPTION
**Pull request by nilsnolde**
fixes #93 

I also upgraded the Github Actions to their latest releases.

### CODE REVIEW

**Comment by nilsnolde** [comment]
Can't see how that wouldn't work, but we won't know before we merge to master. I'd take the risk personally, but if you want I can also try `cibuildwheel` locally, I'm pretty sure I did that before on linux. What's your take @jonathf ?

File: .github/workflows/main_push.yml [hunk 1]
```diff
@@ -45,14 +45,14 @@ jobs:
       if: matrix.platform == 'windows' && steps.cache-conan.outputs.cache-hit != 'true'
       run: |
         pip install pip --upgrade
-        pip install conan==1.59.0
+        pip install conan<2.0.0
```
Comment (nilsnolde) [inline .github/workflows/main_push.yml:48]: they kept releasing 1.x after 2.x was released, they're at 1.61 or so now

Comment (nilsnolde) [reply .github/workflows/main_push.yml:48]: still loathing to invest that one hour to move any of my projects to 2.x, but I guess the time will come eventually

**Comment by jonathf** [comment]
To build without release, use the `-dev` tag suffix. I just pushed the tag `1.13.3-dev5` to demonstrate.

When it builds and deploys properly, you can install it from pypi using the specific tag. E.g. `pip install pyvroom==1.13.3.dev5`. (Note the change in syntax.)

**Comment by nilsnolde** [comment]
Was failing a bit, good idea with -devx release. I'll let you know after it built properly.

File: pyproject.toml [hunk 2]
```diff
@@ -26,12 +26,13 @@ test-command = 'python -c "import vroom"'
...
 archs = "native"
+manylinux-x86_64-image = "quay.io/pypa/manylinux_2_28_x86_64"
 
 [tool.cibuildwheel.linux]
 before-all = """
-yum update -y
-yum install -y epel-release
-yum install -y openssl-devel asio-devel
+dnf update -y
+dnf module enable -y mariadb-devel
```
Comment (nilsnolde) [inline pyproject.toml:34]: weirdest thing: https://bugs.almalinux.org/view.php?id=208&nbn=3

**Comment by nilsnolde** [comment]
this is good to merge, tested the README examples locally with `1.13.3-dev6`

**Comment by nilsnolde** [comment]
Here the action for the dev release: https://github.com/VROOM-Project/pyvroom/actions/runs/5927050551

**Comment by jonathf** [comment]
`1.13.3-dev6` passes the test on my machine as well. Good work, Nils!

**Pull request merged**